Resources/config/mobile_detect.xml

```diff
@@ -17,6 +17,7 @@
         <service id="mobile_detect.mobile_detector.default" class="%mobile_detect.mobile_detector.class%"/>
 
         <service id="mobile_detect.device_view" class="%mobile_detect.device_view.class%">
+            <argument>%mobile_detect.redirect%</argument>
```

Comment:
@orions you need passed `%mobile_detect.redirect%` conf as second argument.
Your code
```
...
public function __construct(RequestStack $requestStack = null, $configRedirect = null)
...
```